File: java/src/localsearchoptimization/parameters/LocalDescentParameters.java
package localsearchoptimization.parameters;

import localsearchoptimization.components.Operator;

public class LocalDescentParameters extends OptimizationParameters {

    public boolean isSteepestDescent = false;

    public Operator[] operators;

    public LocalDescentParameters() {

    }

    protected LocalDescentParameters(LocalDescentParameters copy) {
        super(copy);
        isSteepestDescent = copy.isSteepestDescent;
        operators = copy.operators;
    }

    @Override
    public OptimizationParameters Clone() {
        return new LocalDescentParameters(this);
    }
}

File: C#/LocalSearchOptimization/Parameters/LocalDescentParameters.cs
using LocalSearchOptimization.Components;
using System.Collections.Generic;

namespace LocalSearchOptimization.Parameters
{
    public class LocalDescentParameters : OptimizationParameters
    {
        public bool IsSteepestDescent { get; set; } = false;

        public List<Operator> Operators { get; set; }

        public LocalDescentParameters()
        {

        }

        protected LocalDescentParameters(LocalDescentParameters copy) : base(copy)
        {
            IsSteepestDescent = copy.IsSteepestDescent;
            Operators = copy.Operators;
        }

        public override OptimizationParameters Clone()
        {
            return new LocalDescentParameters(this);
        }
    }
}